adapters/grid/grid.go

```diff
@@ -15,11 +15,53 @@ type GridAdapter struct {
 	endpoint string
 }
 
+func processImp(imp *openrtb.Imp) error {
+	// get the grid extension
+	var ext adapters.ExtImpBidder
+	var gridExt openrtb_ext.ExtImpGrid
+	if err := json.Unmarshal(imp.Ext, &ext); err != nil {
+		return err
+	}
+	if err := json.Unmarshal(ext.Bidder, &gridExt); err != nil {
+		return err
+	}
+
+	if gridExt.Uid == 0 {
+		err := &errortypes.BadInput{
+			Message: "uid is empty",
+		}
+		return err
+	}
+	// no error
+	return nil
+}
+
 // MakeRequests makes the HTTP requests which should be made to fetch bids.
```

Comment:
Can we add a couple more tests in `supplemental` to cover the if statements shown in red below?
![image](https://user-images.githubusercontent.com/15863840/75910357-bf031680-5e1b-11ea-90fd-212dd8799f94.png)